tests/test_qiskit_device.py

```diff
@@ -167,22 +173,46 @@ def test_transpilation_option_update(self, device, wires, device_options, transp
 class TestAnalyticWarningHWSimulator:
     """Tests the warnings for when the analytic attribute of a device is set to true"""
 
-    def test_warning_raised_for_hardware_backend_analytic_expval(self, recorder):
+    testcases_warning_raised_for_hardware_backend_analytic_expval = [
+        (
+            "qiskit.aer",
+            "aer_simulator",
+            (
+                "The analytic calculation of expectations, variances and "
+                "probabilities is only supported on statevector backends, not on the aer_simulator. "
+                "Such statistics obtained from this device are estimates based on samples."
+                "The simulation will be run with shots=1024."
+            ),
+        ),
+        (
+            "qiskit.basicsim",
+            "basic_simulator",
+            (
+                "The plugin does not currently support analytic calculation of expectations, variances "
+                "and probabilities with the BasicProvider backend basic_simulator. Such statistics obtained from this "
+                "device are estimates based on samples. The simulation will be run with shots=1024."
+            ),
+        ),
+    ]
+
+    @pytest.mark.parametrize(
+        "device_backend_msg",
+        testcases_warning_raised_for_hardware_backend_analytic_expval,
+    )
+    def test_warning_raised_for_hardware_backend_analytic_expval(
+        self, device_backend_msg, recorder
+    ):
```

Comment:
Not sure if this is best practice since it's kinda messy. Just didn't want to have (nearly) identical tests for two devices 😅